prow/pjutil/filter.go

```diff
@@ -25,7 +25,8 @@ import (
 	"k8s.io/test-infra/prow/config"
 )
 
-var TestAllRe = regexp.MustCompile(`(?m)^/test all,?($|\s.*)`)
+// TestAllRe provides the regex for `/test all` or `/test <test_name>`.
+var TestAllRe = regexp.MustCompile(`(?m)^(\/test \b[a-zA-Z0-9-_]+\b)$`)
```

Comment:
This is very misleading and will also break other uses of the regexp. I don't understand why it would make sense for the regexp used to identify `/test all` to match things like `/test pull-foo-bar` since `/test pull-foo-bar` does not mean "trigger all required presubmits".